### PR TITLE
CM: Add e2e tests for layout updates

### PR DESCRIPTION
**Pull request by gu-stav**
### What does it do?

Adds end-to-end tests for https://github.com/strapi/strapi/pull/12577.

### Why is it needed?

Because the case it not obvious and it will help in the future to avoid this error.

### How to test it?

Trust the test.

### Related issue(s)/PR(s)

- https://github.com/strapi/strapi/pull/12577

### CODE REVIEW

File: packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      url: '/content-manager/content-types/api::article.article/configuration',
+      method: 'GET',
+    });
+
+    expect(body.data.contentType.layouts.edit).toStrictEqual(FIXTURE_DEFAULT_LAYOUT);
+  });
+
+  test('Update field size', async () => {
+    const payload = [...FIXTURE_DEFAULT_LAYOUT];
+    payload[0][0].size = 12;
```
Comment (petersg83) [inline packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js:68]: I think it still modify `FIXTURE_DEFAULT_LAYOUT`

File: packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        },
+      },
+    });
+
+    const { body } = await rq({
+      url: '/content-manager/content-types/api::article.article/configuration',
+      method: 'GET',
+    });
+
+    expect(body.data.contentType.layouts.edit[0][0].size).toBe(12);
```
Comment (petersg83) [inline packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js:87]: Maybe you can compare the whole layout to see that it didn't impact the rest of the layout ?

File: packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+  });
+
+  test('Update field size with server restart and invalid JSON size', async () => {
+    const payload = [...FIXTURE_DEFAULT_LAYOUT];
+    payload[0][0].size = 8; // title
+    payload[0][1].size = 4; // date
+    payload[1][0].size = 6; // jsonField
```
Comment (petersg83) [inline packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js:94]: maybe you can put the jsonfield in first line so we see that it is put at the end of the layout? Or create a test for that?

**Comment by gu-stav** [comment]
@petersg83 Thanks for your feedback. I've refactored the tests a bit, so that they are more readable (I hope). It now compares always the whole layout to see that other rows aren't affected.

File: packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js [hunk 4]
```diff
@@ -16,6 +19,14 @@ const restart = async () => {
...
 };
 
+const transformLayout = (layout, partialNewLayout) => {
+  return partialNewLayout.reduce((acc, row, rowIndex) => {
+    acc[rowIndex] = row.map((column, columnIndex) => merge(acc[rowIndex][columnIndex], column));
+
+    return acc;
+  }, layout);
+};
+
```
Comment (petersg83) [inline packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js:29]: is `transformLayout` equivalent to lodash merge function?

Capture d’écran 2022-02-23 à 12 02 15

(I should have deep clone in the second call but it shouldn't change the result)

Comment (petersg83) [reply packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js:29]: Maybe better than `cloneDeep` :
```js
_.merge([], FIXTURE_DEFAULT_LAYOUT, transformation)
```

Comment (gu-stav) [reply packages/core/content-manager/server/tests/content-manager/content-types.test.e2e.js]: That's a very good suggestion. I didn't know `_.merge` can operate on arrays. Very nice. Thanks!

**Review by petersg83 — APPROVED** [review]
It looks good to me ! It feels good to have some tests :)

**Pull request merged**